bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java

```diff
@@ -178,15 +178,23 @@ public Response create(@Parameter(description = "rule data", required = true) Ru
                     .header("Location", "rules/" + URLEncoder.encode(newRule.getUID(), StandardCharsets.UTF_8)).build();
         } catch (IllegalArgumentException e) {
             String errMessage = "Creation of the rule is refused: " + e.getMessage();
-            logger.warn("{}", errMessage);
+            logException(e, errMessage);
             return JSONResponse.createErrorResponse(Status.CONFLICT, errMessage);
         } catch (RuntimeException e) {
             String errMessage = "Creation of the rule is refused: " + e.getMessage();
-            logger.warn("{}", errMessage);
+            logException(e, errMessage);
             return JSONResponse.createErrorResponse(Status.BAD_REQUEST, errMessage);
         }
     }
 
+    private void logException(RuntimeException e, String errMessage) {
+        if (logger.isDebugEnabled()) {
+            logger.warn(errMessage, e);
```

Comment:
```suggestion
            logger.warn("{}", errMessage, e);
```

Reply:
@kaikreuzer does this work?
AFAIK there is no warn​(String format, Object... arguments, Throwable t)

Reply:
Yes, that works.

Reply:
whoops. my bad. thanks for the suggestion